dev.c

```diff
@@ -396,29 +385,20 @@ static ssize_t fuse_dev_do_read(struct fuse_conn *fc, struct file *file,
 	read = fc->queue.r.read;
 	write = smp_load_acquire(&fc->queue.r.write);
 
-	while (read != write) {
-		req = fc->queue.r.requests[read];
-		if (sizeof(req->in) + sizeof(req->pxd_rdwr_in) > remain)
-			break;
-
-		fc->queue.r.requests[read] = NULL;
-		read = (read + 1) & (FUSE_REQUEST_QUEUE_SIZE - 1);
-
-		/* Check if a write request is writing zeroes */
-		if (pxd_detect_zero_writes && (req->in.opcode == PXD_WRITE) &&
-		    req->pxd_rdwr_in.size &&
-		    !(req->pxd_rdwr_in.flags & PXD_FLAGS_SYNC)) {
-			fuse_convert_zero_writes(req);
-		}
-
-		copied_this_time = fuse_copy_req_read(req, iter);
-
-		if (copied_this_time < 0) {
-			request_end(fc, req, -EIO);
-		} else {
-			copied += copied_this_time;
-			remain -= copied_this_time;
+	while (read != write && remain >= sizeof(struct rdwr_in)) {
```

Comment:
Does this also remove the optimization for zeros?

Reply:
no it's moved